compliance_checker/suite.py

```diff
@@ -765,20 +765,43 @@ def load_remote_dataset(self, ds_str):
         :param str ds_str: URL to the remote resource
         """
 
-        if opendap.is_opendap(ds_str):
-            return Dataset(ds_str)
+
+        # Some datasets do not support HEAD requests!  The vast majority will,
+        # however, support GET requests
+        try:
+            head_req = requests.head(ds_str, allow_redirects=True, timeout=10)
+            head_req.raise_for_status()
+        except:
+            warnings.warn("Could not complete HEAD request for resource "
+                          "{}.  Skipping content-type headers".format(ds_str))
+            content_type = None
         else:
+            content_type = head_req.headers.get("content-type")
+
+        # if the Content-Type header returned was "application/x-netcdf",
+        # or a netCDF file (not OPeNDAP) we can open this into an
+        if content_type == "application/x-netcdf":
+            response = requests.get(ds_str, allow_redirects=True,
+                                    timeout=60)
+            # TODO: handle case when netCDF C libs weren't compiled with
+            # in-memory support by using tempfiles or other means
+            return MemoizedDataset(response.content, memory=response.content)
+
+        elif opendap.is_opendap(ds_str):
+            return Dataset(ds_str)
```

Comment:
Should we use the MemoizedDataset form here?  It's potentially faster if we know the data isn't changing underneath us.   I don't fully understand the OPeNDAP protocol though, so I'm leaving this as is for now.